test/test_fx_experimental.py

```diff
@@ -0,0 +1,60 @@
+import torch
+
+from torch.fx.symbolic_trace import symbolic_trace
+from torch.fx.experimental import GraphManipulation
+from torch.fx.experimental.Partitioner import Partitioner, DAG
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.jit_utils import JitTestCase
+
+
+class TestFX(JitTestCase):
+    def test_size_based_partition(self):
+        class testModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear_mod_0 = torch.nn.Linear(3, 4)
+                self.linear_mod_1 = torch.nn.Linear(4, 4)
+            def forward(self, a, b, c):
+                d = self.linear_mod_0(a)
+                e = self.linear_mod_1(b)
+                f = torch.rand(4)
+                g = torch.rand(5)
+                return c + d + e, f, g
+
+        m = testModule()
+        traced = symbolic_trace(m)
+        GraphManipulation.get_size_of_all_nodes(
+            traced,
+            [torch.rand(3), torch.rand(4), torch.rand(4)]
+        )
+        partitioner = Partitioner()
+        devices = [
+            {'name': 'dev_0', 'available_mem': 1000},
+            {'name': 'dev_1', 'available_mem': 1000},
+            {'name': 'dev_2', 'available_mem': 1000}
+        ]
+        dag = partitioner.partition_graph(traced, devices)
+        nodes = traced.graph.nodes
+        ref_dag = DAG()
+        ref_dag.create_node(0, [], [1, 2], [], [])
+        ref_dag.create_node(1, [0], [3], [nodes[1]], [nodes[4]])
+        ref_dag.create_node(2, [0], [3], [nodes[0], nodes[7]], [nodes[3], nodes[7]])
+        ref_dag.create_node(3, [1, 2], [], [nodes[2], nodes[8], nodes[4], nodes[3]], [nodes[6], nodes[8]])
```

Comment:
I'm worried this test is going to be very fragile if the representation of code in `GraphModule`s changes even slightly. One idea I want to explore is whether `Partitions` can have generated, runnable code (e.g. if `Partition` is a subclass of `GraphModule`). Then we can test correctness by running the Partitions, rather than hooking into minute details like this